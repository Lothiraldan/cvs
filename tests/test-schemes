#!/bin/sh

cat <<EOF >> $HGRCPATH
[extensions]
schemes=

[schemes]
l = http://localhost:$HGPORT/
parts = http://{1}:$HGPORT/
z = file:\$PWD/
EOF

hg init test
cd test
echo a > a
hg ci -Am initial

hg serve -n test -p $HGPORT -d --pid-file=hg.pid -A access.log -E errors.log
cat hg.pid >> $DAEMON_PIDS

hg incoming l://

echo % check that {1} syntax works
hg incoming --debug parts://localhost | sed 's/[0-9]//g'

echo % check that paths are expanded
PWD=`pwd` hg incoming z://

echo % errors
cat errors.log
