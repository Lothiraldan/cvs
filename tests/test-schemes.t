#require serve

  $ cat <<EOF >> $HGRCPATH
  > [extensions]
  > schemes=
  > 
  > [schemes]
  > l = http://localhost:$HGPORT/
  > parts = http://{1}:$HGPORT/
  > z = file:\$PWD/
  > EOF
  $ hg init test
  $ cd test
  $ echo a > a
  $ hg ci -Am initial
  adding a

invalid scheme

  $ hg log -R z:z
  abort: no '://' in scheme url 'z:z'
  [255]

http scheme

  $ hg serve -n test -p $HGPORT -d --pid-file=hg.pid -A access.log -E errors.log
  $ cat hg.pid >> $DAEMON_PIDS
  $ hg incoming l://
  comparing with l://
  searching for changes
  no changes found
  [1]

check that {1} syntax works

  $ hg incoming --debug parts://localhost
  using http://localhost:$HGPORT/
  sending capabilities command
  comparing with parts://localhost/
  query 1; heads
  sending batch command
  searching for changes
  all remote heads known locally
  no changes found
  [1]

check that paths are expanded

  $ PWD=`pwd` hg incoming z://
  comparing with z://
  searching for changes
  no changes found
  [1]

check that debugexpandscheme outputs the canonical form

  $ hg debugexpandscheme bb://user/repo
  https://bitbucket.org/user/repo

expanding an unknown scheme emits the input

  $ hg debugexpandscheme foobar://this/that
  foobar://this/that

expanding a canonical URL emits the input

  $ hg debugexpandscheme https://bitbucket.org/user/repo
  https://bitbucket.org/user/repo

errors

  $ cat errors.log

  $ cd ..
