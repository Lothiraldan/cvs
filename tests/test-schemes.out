adding a
comparing with l://
searching for changes
no changes found
% check that {1} syntax works
using http://localhost:/
sending between command
comparing with parts://localhost
sending heads command
searching for changes
no changes found
% check that paths are expanded
comparing with z://
searching for changes
no changes found
% errors
